packages/expect/src/__tests__/matchers.test.js

```diff
@@ -271,6 +271,16 @@ describe('.toStrictEqual()', () => {
     expect(c.constructor.name).toEqual(d.constructor.name);
     expect({test: c}).not.toStrictEqual({test: d});
   });
+
+  it('does not pass when sparseness of arrays do not match', () => {
+    expect([, 1]).not.toStrictEqual([undefined, 1]); // eslint-disable-line no-sparse-arrays
+    expect([undefined, 1]).not.toStrictEqual([, 1]); // eslint-disable-line no-sparse-arrays
+    expect([, , , 1]).not.toStrictEqual([, 1]); // eslint-disable-line no-sparse-arrays
+  });
+
+  it('does not pass when equally sparse arrays have different values', () => {
+    expect([, 1]).not.toStrictEqual([, 2]); // eslint-disable-line no-sparse-arrays
+  });
```

Comment:
do we have a test that `expect([, 1]).toStrictEqual([, 1]);` as well?

Reply:
Done in https://github.com/facebook/jest/pull/7591/commits/fe65db71ee364b4516eb617a851601aace5393aa